dotnet-authserver/src/TeacherIdentity.AuthServer/AuthenticationState.cs

```diff
@@ -335,6 +322,11 @@ public void OnMobileNumberVerified(User? user = null)
         }
 
         MobileNumberVerified = true;
+
+        if (user is not null)
+        {
+            UpdateAuthenticationStateWithUserDetails(user);
```

Comment:
If email address entered in previous step is different to existing user address do we need to redirect to register/choose-email page?

Reply:
Also assuming we want to do similar sign-in for register/email-confirmation

Reply:
There are/will be separate 'edge case' tickets for dealing with both of those.